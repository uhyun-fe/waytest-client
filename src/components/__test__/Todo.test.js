import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "../Todo";

afterEach(() => {
   cleanup();
});

test("should render none-completed todo", () => {
   const todo = { id: 1, title: "할일 111", completed: false };
   render(<Todo todo={todo} />);
   // const todoElement = screen.getByTestId("todo-1");
   // expect(todoElement).toBeInTheDocument();
   // expect(todoElement).toHaveTextContent("할일 111");
   // expect(!todoElement.innerHTML.includes("<strike>")).toBeTruthy();
});

// test("should render completed todo", () => {
//    const todo = { id: 2, title: "할일 222", completed: true };
//    render(<Todo todo={todo} />);
//    const todoElement = screen.getByTestId("todo-2");
//    expect(todoElement).toBeInTheDocument();
//    expect(todoElement).toHaveTextContent("할일 222");
//    expect(todoElement.innerHTML.includes("<strike>")).toBeTruthy();
// });

// test("matches snapshot", () => {
//    const todo = { id: 1, title: "할일 111", completed: false };
//    const tree = renderer.create(<Todo todo={todo} />).toJSON();
//    expect(tree).toMatchSnapshot();
// });
