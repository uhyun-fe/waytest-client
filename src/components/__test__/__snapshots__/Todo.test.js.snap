// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    할일 111
  </h1>
</div>
`;
